### PR TITLE
attempt to support windows

### PR DESCRIPTION
**Pull request by tamird**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/c-rocksdb/9)

### CODE REVIEW

**Comment by bdarnell** [comment]
How have you tested this? If we're going to attempt to support windows, appveyor.com is a good windows-based CI platform (and you can set up multiple CI systems with github; I do this for Tornado). 

However, this seems like an unnecessary distraction as we're approaching beta and we should wait to explore this until post-beta.

**Comment by tamird** [comment]
Yes, I agree. I was trying to get cross-compilation working using https://github.com/karalabe/xgo, and wanted to document what I had tried. I won't be actively working on this.

**Comment by tamird** [comment]
I should mention that this currently fails in xgo: https://github.com/karalabe/xgo/issues/26

File: cgo_flags.go [hunk 1]
```diff
@@ -6,9 +6,10 @@ package rocksdb
 // #cgo CPPFLAGS: -Iinternal -Iinternal/include -Iinternal/db -Iinternal/util
 // #cgo CPPFLAGS: -Iinternal/utilities/merge_operators/string_append
 // #cgo CPPFLAGS: -I../c-snappy/internal -I../c-lz4/internal/lib
-// #cgo CPPFLAGS: -DROCKSDB_PLATFORM_POSIX -DNDEBUG -DSNAPPY -DLZ4
-// #cgo darwin CPPFLAGS: -DOS_MACOSX
-// #cgo linux CPPFLAGS: -DOS_LINUX
+// #cgo CPPFLAGS: -DNDEBUG -DSNAPPY -DLZ4
+// #cgo darwin CPPFLAGS: -DOS_MACOSX -DROCKSDB_PLATFORM_POSIX
+// #cgo linux CPPFLAGS: -DOS_LINUX -DROCKSDB_PLATFORM_POSIX
+// #cgo windows CPPFLAGS: -DOS_WIN
```
Comment (petermattis) [inline cgo_flags.go:9]: Is someone asking for windows support? Just curious what motivated this.

Comment (tamird) [inline cgo_flags.go:9]: Nah, I was trying to get cross-compilation working so we could post binaries and wanted to see how far I could get with windows.

**Comment by petermattis** [comment]
LGTM

**Comment by tamird** [comment]
Hey, this is ready for another look. For now, I've worked around some issues by omitting snappy on windows. This compiles successfully with `xgo -go 1.6.1 --targets=windows-6.0/amd64 .`.

**Comment by tbg** [comment]
:lgtm:, but with a healthy dose of science dog.

<details><summary>

<i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>



> Hey, this is ready for another look. For now, I've worked around some issues by omitting snappy on windows. This compiles successfully with `xgo -go 1.6.1 --targets=windows-6.0/amd64 .`.
> </details>

---

Reviewed 6 of 8 files at r1, 6 of 6 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/c-rocksdb/9#-:-KIiIgDAEciGoq--UMy4:1173689132)_

**Pull request merged**